updater/proxy.go

```diff
@@ -18,23 +18,23 @@ import (
 
 func setupHTTPProxy() {
 	// Force if set via flag
-	if len(*httpProxy) > 0 {
-		_ = os.Setenv("HTTP_PROXY", *httpProxy)
-		return
-	}
-	// Check Silver Environment
-	proxy := os.Getenv("SILVER_HTTP_PROXY")
-	if proxy != "" {
-		_ = os.Setenv("HTTP_PROXY", proxy)
-		return
-	}
-	// Proxy conf file
-	if dat, err := ioutil.ReadFile("http-proxy.conf"); err == nil {
-		proxy = strings.TrimSpace(string(dat))
+	proxy := *httpProxy
+
+	if proxy == "" {
+		// Else check Silver Environment
+		proxy = os.Getenv("SILVER_HTTP_PROXY")
+		if proxy == "" {
+			// Else check proxy conf file
+			if dat, err := os.ReadFile("http-proxy.conf"); err == nil {
+				proxy = strings.TrimSpace(string(dat))
+			}
+		}
 	}
+
 	if proxy != "" {
+		fmt.Printf("Using proxy: %s\n", proxy)
```

Comment:
This should be shown after verifying an error returned from `os.Setenv`.
Also, would it be better to indicate "HTTP"?